src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,272 @@
+    let superClass = this.transformAny(tree.superClass);
+
+    statements = prependStatements(statements,
+        createClassDeclaration(tree.name, superClass, elements));
+
+    return new AnonBlock(null, statements);
+  }
+
+  /**
+   * Transforms a single class expression
+   *
+   * @param {ClassExpression} tree
+   * @return {ParseTree}
+   */
+  transformClassExpression(tree) {
+    let {
+      elements,
+      initStaticVars,
+    } = this.transformClassElements_(tree);
+
+    let superClass = this.transformAny(tree.superClass);
+    let classExpression = new ClassExpression(tree.location, tree.name,
+        superClass, elements, tree.annotations, tree.typeParameters);
+
+    if (initStaticVars.length === 0) {
+      return classExpression;
+    }
+
+    this.pushTempScope();
+    let id = this.getTempIdentifier();
+    let className = createBindingIdentifier(id);
+    let statements = [
+      parseStatement `let ${id} = ${classExpression}`,
+      ...createStaticInitializerStatements(className, initStaticVars),
+      parseStatement `return ${className}`
+    ];
+    let body = createFunctionBody(statements);
+    this.popTempScope();
+
+    return createImmediatelyInvokedFunctionExpression(body);
+  }
+
+
+  getDefaultConstructor_(tree, initStatements) {
+    let constructorParams = createEmptyParameterList();
+    let constructorBody;
+    if (tree.superClass) {
+      let statement = parseStatement `super(...arguments)`;
+      constructorBody = createFunctionBody([statement, ...initStatements]);
+    } else {
+      constructorBody = createFunctionBody(initStatements);
+    }
+
+    let name = new LiteralPropertyName(null, createIdentifierToken(CONSTRUCTOR));
+
+    return new PropertyMethodAssignment(tree.location, false,
+        null, name, constructorParams, null, [], constructorBody);
+  }
+
+  /**
+   *  Appends assignment to this for the member variables after the super call.
+   */
+  appendInstanceInitializers_(constructor, initStatements, superClass) {
+    if (initStatements.length === 0) return constructor;
+
+    let statements = constructor.body.statements;
+
+    if (superClass) {
+      let superExpressionIndex = -1;
+
+      for (let i = 0; i < statements.length; i++) {
+        let statement = statements[i];
+        if (statement.isDirectivePrologue()) {
+          continue;
+        }
+        if (isSuper(statement)) {
+          superExpressionIndex = i;
+          break;
+        }
+      }
+
+      statements = statements.slice();
+      statements.splice(superExpressionIndex + 1, 0, ...initStatements);
+    } else {
+      statements = prependStatements(statements, ...initStatements);
+    }
+
+    return new PropertyMethodAssignment(constructor.location, false,
+        constructor.functionKind, constructor.name, constructor.parameterList,
+        constructor.typeAnnotation, constructor.annotations,
+        createFunctionBody(statements));
+  }
+}
+
+// TODO(vicb): Does not handle computed properties
```

Comment:
Now that it is using defineProperty it is trivial to support computed properties.